### DISCUSSION TITLE
Snapshot + Bitrate take insane amount of CPU

### DISCUSSION THREAD

**Discussion by riftdea-isj** [ant-media/Ant-Media-Server#5677] [Q&A]
Hi Antmedia Team, I'm here looking for suggestion, in the situation of 100 stream with bitrate enable and snapshot enabled... 40CPU VM is at constant of 90% CPU usage, its insanely high... Is there a way to optimized this via antmedia itself, if not is there a tools that I can intergrade with antmedia on a separate server from antmedia-server to do the **snapshot**... Would be best if able to share the CPU load.

Enterprise Version

**Comment by lastpeony** [comment]
Sounds like you need to scale your ant media server so that load can be distributed across cluster nodes.
https://antmedia.io/docs/category/clustering-and-scaling/
Also consider enabling GPU if you have many ABR's enabled.
https://antmedia.io/docs/guides/advanced-usage/using-nvidia-gpu/

**Reply by riftdea-isj** [reply]
Yes but I would like to have a actually separated server that can do the snapshot task instead all in AM server. 
Any other tools that can integrate with AM, on a separate server. Apparently, with snapshot on, it spike the CPU too much from time to time and resulting of stream dropping.

**Reply by lastpeony** [reply]
What do you mean by snapshot? Do you mean taking snapshot of the disk?
I believe i couldnt get your question.

**Reply by riftdea-isj** [reply]
I apologized, I mean the generate preview.

**Reply by lastpeony** [reply]
no problem
utilizing gpu should solve high cpu usage

**Reply by riftdea-isj** [reply]
Yeah that's a way to do it, but I'm trying to explore purely on the software side. Is there a way that I can offload the AM server, any software that can help generate preview that can be integrated into AM, from a separate server.

**Reply by lastpeony** [reply]
you can play the stream on a cloud server in chrome and periodicly take screenshot using selenium.

**Reply by mekya** [reply]
Hi @riftdea-isj @lastpeony 
I just want to add that you can use another server to generate previews from HLS streams with ffmpeg. It's a better approach in terms of cost and performance.

https://trac.ffmpeg.org/wiki/Create%20a%20thumbnail%20image%20every%20X%20seconds%20of%20the%20video

Does it work for you?
